index.js

```diff
@@ -13,6 +13,7 @@ module.exports = (filename, options) => {
 
 	options = {
 		compress: true,
+		streamOutput: false,
```

Comment:
Why make it an option? Could you not check `file.isStream()` and just do it if `true`?

Reply:
I also originally wondered that but to explain my thinking here:
1) We could have buffers and streams coming in, so we'd only be basing that decision off the first one we see
2) To me, wanting to stream files doesn't imply wanting the result as a stream as well, mainly since streams are not as well supported in the gulp ecosystem and may cause problems down the line
3) I think having this as a default-off option would be a minor change, whereas taking the setting from the input files could be breaking since it would change the output from a buffer to a stream wherever streams are coming in

I'm happy to change it if you think I'm missing the mark there though.

Reply:
1. I don't think that's possible. A Gulp pipeline is either fully streamed or not at all, since even if only a single plugin in a pipeline doesn't support streaming, it will have to buffer for that plugin and it's all moot.
3. I'm ok with doing a major release for this.

Reply:
I think it's uncommon but definitely possible to have multiple modes on the Vinyl stream aside from plugin support. This example is a bit contrived, but I could do:

```js
const mergeStream = require('merge-stream');
const gulp = require('gulp');
const gulpZip = require('gulp-zip');

exports.default = function() {
  return mergeStream(
    gulp.src('*.mp4', { buffer: false }),
    gulp.src('*.txt'),
  )
  .pipe(gulpZip())
  .pipe(gulp.dest('out/myzip.zip'));
}
```

I think I'd have a race condition where I'll either get a streamed or a buffered zip depending on which file arrives first, which I don't think should happen, and that also means can't force streaming on if I'm having the oversized buffer problem.

What about instead if I changed it so that an oversized buffer error automatically switches the output to streaming and I'll add a note about it to the readme? Then we could avoid the option and automatically handle the error.

Reply:
We cannot automatically switch to streaming as it affects the plugins coming after this one.

I think you should rather bring this up on the Gulp/Vinyl issue tracker. Plugins should not be opinionated about this. They should support either buffer or both and adhere to the mode Gulp tells them.

Reply:
Sorry for the long delay on responding here. Yeah I agree that auto switching is the wrong way to go and would be confusing.

The problem I have with doing it automatically is that the file contents being buffered or streams is a property of each Vinyl file, but not of the stream as a whole. There's no API to check the streaming/buffered mode of the whole stream and it's possible to construct valid streams with mixed content, so my interpretation is that this is an intentional and quite useful feature of Gulp. I believe that plugins need to respect the mode of each file in the stream and it's not correct to infer the nature of every file in the stream from one instance. If I change this to automatically switch mode based on the first file it will introduce inconsistent behavior for mixed content streams like the one I mentioned above (the output could alternate between streaming and buffered depending on which file comes through the pipe first).

I think that given that, my opinion is that an option is really the only consistent and safe way to handle this. It's unambiguous and discoverable and gives the users of the plugin the choice without introducing the potential for inconsistent behaviour.

Reply:
Seems like an option is the only choice, yes, but I also think your use-case is very obscure, and that it would be easier for you to just make everything streaming. 🤷🏻‍♂️

Reply:
Yeah it isn't popular to do that, but I think that's a shame because in my opinion it's one of the coolest Gulp/Vinyl features that the API is based on regular streams and for example merging and splitting can be done at the stream level without Gulp even needing to know or provide APIs for it. I think it's a very powerful feature! That's why I think it's worth us supporting it with this change.

So if I put the option back in would you be ok with that?

Reply:
> So if I put the option back in would you be ok with that?

Yes, but it should be clearly documented in what case the option is useful.

Reply:
Ok the option is back but renamed to `buffer` to match the `gulp.src` `buffer` option which I hope should make it easy to interpret along with the docs.